esphome/components/emporia_vue/emporia_vue.h

```diff
@@ -39,39 +39,48 @@ class CTSensor;
 
 class EmporiaVueComponent : public Component, public i2c::I2CDevice {
  public:
-  void setup() override;
-  void loop() override;
   void dump_config() override;
-  static void i2c_request_task(void *pv);
 
+  void set_sensor_poll_interval(uint32_t sensor_poll_interval) { this->sensor_poll_interval_ = sensor_poll_interval; }
+  uint32_t get_sensor_poll_interval() const { return this->sensor_poll_interval_; }
   void set_phases(std::vector<PhaseConfig *> phases) { this->phases_ = phases; }
   void set_ct_sensors(std::vector<CTSensor *> sensors) { this->ct_sensors_ = sensors; }
 
+  void setup() override;
+  void loop() override;
+
  private:
+  static void i2c_request_task(void *pv);
+
+  uint32_t sensor_poll_interval_;
   std::vector<PhaseConfig *> phases_;
   std::vector<CTSensor *> ct_sensors_;
   QueueHandle_t i2c_data_queue_;
 };
 
-enum PhaseInputWire : uint8_t { 
+enum PhaseInputWire : uint8_t {
   BLACK = 0,
   RED = 1,
-  BLUE = 2, 
+  BLUE = 2,
 };
 
 class PhaseConfig {
  public:
   void set_input_wire(PhaseInputWire input_wire) { this->input_wire_ = input_wire; }
-  void set_calibration(double calibration) {this->calibration_ = calibration; }
-  void set_voltage_sensor(sensor::Sensor voltage_sensor) {this->set_voltage_sensor = voltage_sensor; }
-  double get_calibration() { return this->calibration_; }
-  void update_voltage(EmporiaSensorData &data);
-  int32_t extract_power_for_phase(const PowerDataEntry &entry);
+  PhaseInputWire get_input_wire() const { return this->input_wire_; }
+  void set_calibration(float calibration) { this->calibration_ = calibration; }
+  float get_calibration() const { return this->calibration_; }
```

Comment:
I believe we are getting a float from Python, and I don't think it's worth getting any more precise than that here, so I changed this from a double to a float

Reply:
Sounds good, I originally put it as a double since the float precision was changing the value a bit (because of the mantissa length) but I don't think it will affect the value that much in the end.